ngraph/test/type_prop/gather_nd.cpp

```diff
@@ -211,4 +318,20 @@ TEST(type_prop, gather_nd_fail_indices_element_type)
     {
         FAIL() << "Deduced type check failed for unexpected reason";
     }
+    NGRAPH_SUPPRESS_DEPRECATED_END
+
+    try
+    {
+        auto G5 = make_shared<op::v5::GatherND>(P, I);
+        // Should have thrown, so fail if it didn't
+        FAIL() << "Incorrect indices element type";
+    }
+    catch (const NodeValidationFailure& error)
+    {
+        EXPECT_HAS_SUBSTRING(error.what(), std::string("The indices type is expected to be an integer type."));
+    }
+    catch (...)
+    {
+        FAIL() << "Deduced type check failed for unexpected reason";
+    }
```

Comment:
I think you should add some tests to confirm shape inference for cases when `data` and `indices` shapes are not fully known

Reply:
Agree. I will add with dynamic dimensions. Thanks

Reply:
@mbencer, I have added three more advanced cases with dynamic dimensions.